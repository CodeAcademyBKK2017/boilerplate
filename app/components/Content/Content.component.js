// import Icon from 'react-native-vector-icons/Foundation';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Content.style';
import {Text, TextInput, View} from 'react-native';

export default class TextArea extends Component {
  render () {
    return (
      <View style={styles.container}>
        <View style={styles.headContent}>
          <Text style={styles.text}>Please type your note below</Text>
          {/* <Icon name='clipboard-notes' size={20} /> */}
        </View>
        <TextInput style={styles.textArea} 
          multiline={true} 
          underlineColorAndroid='transparent'
          onChangeText={this.props.onContentChange}  
          value={this.props.value}/>
      </View>
    );
  }
}
TextArea.propTypes = {
  onContentChange: PropTypes.func,
  value: PropTypes.string
};
