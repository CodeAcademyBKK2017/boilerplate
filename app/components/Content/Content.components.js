import React, {Component} from 'react';
import {

  StyleSheet,
  Text,
  TextInput,
  View
} from 'react-native';

class Content extends Component {
  render () {
    return (
      <View style={styles.conView}>
        <Text style={styles.welcome}>
         Please type your note below
        </Text>
        <TextInput
          style={styles.textin}
          // onChangeText={this.state.nans}
          multiline = {true}
          underlineColorAndroid = 'transparent'
          numberOfLines = {4}   
          placeholder='-Understand how react-native works.'
        />
      </View>
    );
  }
}
const styles = StyleSheet.create({
  welcome: {
    fontSize: 12,
    // textAlign: 'center',
    margin: 10
   
  },
  conView: {
    flex: 1,
    marginTop: 10
  },
  textin: {
    // width: 300,
    // height: 300,
    borderColor: 'gray',
    borderWidth: 1,
    flex: 1,
    backgroundColor: '#fff'

  }
});
export default Content;