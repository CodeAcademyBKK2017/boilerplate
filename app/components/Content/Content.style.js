import React, {StyleSheet} from 'react-native';
  
export default StyleSheet.create({
  welcome: {
    fontSize: 16,
    textAlign: 'center',
    margin: 10,

    fontStyle: 'italic'
  },
  textArea: {
    width: '100%',
    borderColor: 'gray',
    borderWidth: 1,
    backgroundColor: '#fff',
    padding: 5,
    color: '#ababab',
    borderRadius: 3
  },
  content: {
    flexDirection: 'row',
    margin: 20
  },
  conArea: {
    flex: 1,
    flexDirection: 'row',
    marginLeft: 30,
    marginRight: 30
  },
  title: {
    flex: 1,
    flexDirection: 'row',
    margin: 30
  },
  noteLeft: {
    flex: 1
  },
  noteRight: {
    flex: 0
  },
  note: {
    fontWeight: '600'
  }
});