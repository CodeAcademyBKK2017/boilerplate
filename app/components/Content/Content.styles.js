import React, {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    width: '100%',
    flex: 1,
    paddingBottom: 10
  },
  title: {
    fontSize: 13,
    color: '#444',
    fontStyle: 'italic'
  },
  textInput: {
    flex: 1,

    marginVertical: 10,
    paddingVertical: 15,
    paddingHorizontal: 12,

    borderWidth: 1,
    borderColor: '#c4c4c5',
    borderRadius: 3,
    
    backgroundColor: '#ffffff',
    textAlignVertical: 'top'
  }
});