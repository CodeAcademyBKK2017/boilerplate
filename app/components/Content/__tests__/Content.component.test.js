import 'react-native';
import Content from '../Content.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Content', () => {
  it('snapshot test', () => {
    const snapshot = renderer.create(<Content />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});