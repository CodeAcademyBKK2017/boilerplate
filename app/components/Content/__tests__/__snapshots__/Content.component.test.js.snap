// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontStyle": "italic",
          "paddingVertical": 10,
        }
      }
    >
      Please type your note below
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 18,
          },
          Object {
            "padding": 10,
          },
          Object {
            "fontFamily": "fontcustom",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï„¨
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "gray",
        "borderWidth": 1,
        "flex": 1,
        "padding": 5,
        "textAlignVertical": "top",
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
